internal/sirius/document.go

```diff
@@ -0,0 +1,121 @@
+package sirius
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+type Document struct {
+	ID              int    `json:"id,omitempty"`
+	CorrespondentID int    `json:"correspondentID"`
+	Type            string `json:"type"`
+	SystemType      string `json:"systemType"`
+	FileName        string `json:"fileName,omitempty"`
+	Content         string `json:"content"`
+}
+
+type documentTemplateApiResponse map[string]json.RawMessage
+type insertApiResponse map[string]json.RawMessage
+
+type UniversalTemplateData struct {
+	Location        string `json:"location"`
+	OnScreenSummary string `json:"onScreenSummary"`
+}
+
+type DocumentTemplateApiData struct {
+	Inserts insertApiResponse `json:"inserts"`
+	UniversalTemplateData
+}
+
+type Insert struct {
+	Key      string
+	InsertId string
+	UniversalTemplateData
+}
+
+type DocumentTemplateData struct {
+	Inserts    []Insert
+	TemplateId string
+	UniversalTemplateData
+}
+
+func (d documentTemplateApiResponse) toDocumentData() ([]DocumentTemplateData, error) {
+	var s []DocumentTemplateData
+
+	for k, v := range d {
+		var asDocumentTemplateData DocumentTemplateData
+		asDocumentTemplateData.TemplateId = k
+
+		var asDocumentTemplateApiData DocumentTemplateApiData
+		if err := json.Unmarshal(v, &asDocumentTemplateApiData); err == nil {
```

Comment:
It seems weird to be unmarshalling here having just done so in `DocumentTemplates`. Could you not do:
```go
type documentTemplateApiResponse map[string]DocumentTemplateApiData
```

Reply:
The issue is with the inserts field which will either return an empty array `[]` or a `map[string]json.RawMessage`. So it will throw an error when trying to decode the insert field when it's an empty array